csharp/extractor/Semmle.Extraction.CSharp.Standalone/BuildAnalysis.cs

```diff
@@ -336,7 +336,23 @@ private void DownloadMissingPackages(IEnumerable<string> restoreTargets)
             var alreadyDownloadedPackages = Directory.GetDirectories(packageDirectory.DirInfo.FullName).Select(d => Path.GetFileName(d).ToLowerInvariant()).ToHashSet();
             var notYetDownloadedPackages = new HashSet<string>();
 
-            var allFiles = GetFiles("*.*").ToArray();
+            var nugetConfigs = GetFiles("nuget.config", recurseSubdirectories: true).ToArray();
+            string? nugetConfig = null;
+            if (nugetConfigs.Length > 1)
+            {
+                progressMonitor.MultipleNugetConfig(nugetConfigs);
+                nugetConfig = GetFiles("nuget.config", recurseSubdirectories: false).FirstOrDefault();
+                if (nugetConfig == null)
+                {
+                    progressMonitor.NoTopLevelNugetConfig();
+                }
+            }
+            else
+            {
+                nugetConfig = nugetConfigs.First();
```

Comment:
Should this be `FirstOrDefault()`? (if there is no `nuget.config` then nugetConfigs is empty and `First()` will throw an exception).

Reply:
Thanks.